task_schedule.pl

```diff
@@ -260,6 +262,21 @@
     # Exit if all the cron tasks have been disabled
     exit(0) unless grep { not $_->{disabled} } @crontab;
 
+    ##*****************************************************
+    ##  Check elapsed run time and exit if over threshold
+    ##  (minus a minute)
+    ##*****************************************************
+    my $curr_time = time();
+    my $run_time = $curr_time - $start_time;
+    if ($run_time > (86400 * $opt{max_persistent_days} - 60)){
+        my $days = $run_time / 86400.0;
+        my $message = sprintf(
+                              "Ending this task_schedule; job %.2f days old",
+                              $days);
+        dbg "$message";
+        exit(0);
```

Comment:
> Do we want to remove the heartbeat file at this point? Basically this
> instance of task_schedule is no longer in charge and anyone else is clear
> to start up.
>

I don't think so.  I think of the heartbeat file as marking when the task
ran last, and this N days exit happens after the task as run, so the
heartbeat is still valid.   For the iterations=1 use of task_schedule we
don't delete the heartbeat when done, and I think this is equivalent.  The
reason I added the 60 early pad was to try to remove conflicts, though I
suppose it might have made more sense to subtract off 1 + the timeout or
some such.